### PR TITLE
Optional Sticky toolbar

### PR DESCRIPTION
**Pull request by davide-scalzo**

### CODE REVIEW

File: src/components/MdeToolbar.tsx [hunk 1]
```diff
@@ -8,45 +8,85 @@ export interface MdeToolbarProps {
...
+            window.addEventListener("touchend", this.handleOnScroll);
+        }
+    }
+
+    handleOnScroll() {
+        const scrollTop = window.scrollY;
+        const toolbars = document.getElementsByClassName("mde-header") as HTMLCollectionOf<HTMLElement>;
+        const toolbarScroll = toolbars[0].offsetTop;
+        if (!this.state.scrollHeight) this.setState({ scrollHeight: toolbarScroll });
+        if (scrollTop > toolbarScroll && this.state.scrolled === false && this.props.sticky) {
```
Comment (ericjuta) [inline src/components/MdeToolbar.tsx:43]: Not of a fan of mixing setState and scrolling as I've had performance issues before 

https://www.w3schools.com/howto/howto_js_sticky_header.asp
Coulda also done this

Also you forgot to set this.scrollListenerId + this.touchendListenerId and remove them via componentWillUnmount()

Comment (ericjuta) [reply src/components/MdeToolbar.tsx:43]: Implementation's fine though if it works, only a sticky bar tbh
Just add the componentWillUnmount

**Review by ericjuta — CHANGES_REQUESTED** [review]

File: src/components/MdeToolbar.tsx (hunk 1 above)
Comment (ericjuta) [reply src/components/MdeToolbar.tsx:43]: Looking forward to useEffect hooks as its api is concise for exactly this!

Comment (davide-scalzo) [reply src/components/MdeToolbar.tsx:43]: Done :D

**Review by ericjuta — APPROVED** [review]

**Pull request merged**